polkadot/node/core/prospective-parachains/src/fragment_tree.rs

```diff
@@ -879,6 +882,93 @@ impl FragmentTree {
 		best_result
 	}
 
+	// Orders the ancestors into a viable path from root to the last one.
+	// Returns a pointer to the last node in the path.
+	// If there are any timed out ancestors, trims the path so that we get the largest possible
```

Comment:
This comment is an artifact of an older version of this code. I'll change it.

Currently, when there's a timed out candidate, we just don't add it to the ancestors HashSet in the provisioner.
So, if we have A B C on the cores and B is timed out, ancestor path will contain A and C. So it will suggest new candidates which descend from A. The runtime will evict B and C from the cores